packages/core-mobile/app/services/fcm/FCMService.ts

```diff
@@ -0,0 +1,84 @@
+import messaging, {
+  FirebaseMessagingTypes
+} from '@react-native-firebase/messaging'
+import Logger from 'utils/Logger'
+import NotificationsService from 'services/notifications/NotificationsService'
+import { ChannelId } from 'services/notifications/channels'
+import { ACTIONS, PROTOCOLS } from 'contexts/DeeplinkContext/types'
+
+type UnsubscribeFunc = () => void
+
+/**
+ * Wrapper for @react-native-firebase/messaging
+ */
+class FCMService {
+  getFCMToken = async (): Promise<string> => {
+    return await messaging().getToken()
+  }
```

Comment:
we will also have to do https://rnfirebase.io/reference/messaging#onTokenRefresh to get the updated token and re-register with our server

Reply:
are you planning to do that in a new pr or this pr?